apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/ClusterService.java

```diff
@@ -148,9 +148,6 @@ public void createDefaultCluster(String appId, String createBy) {
       return Collections.emptyList();
     }
 
-    // to make sure parent cluster is ahead of branch cluster
-    Collections.sort(clusters);
```

Comment:
I think this should remain as it is since we need to delete parent cluster earlier than child cluster or the child namespace won't be deleted. see https://github.com/ctripcorp/apollo/blob/master/apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/NamespaceService.java#L281

Reply:
那这样其实删除了父集群就没必要在这里去删除子集群了
https://github.com/ctripcorp/apollo/blob/79e94e715b82f90d9ce82e14621e007253890bc4/apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/AdminService.java#L62

因为如果存在子集群的话 下面已经把子集群的信息都删除了https://github.com/ctripcorp/apollo/blob/79e94e715b82f90d9ce82e14621e007253890bc4/apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/NamespaceService.java#L288

https://github.com/ctripcorp/apollo/blob/79e94e715b82f90d9ce82e14621e007253890bc4/apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/NamespaceBranchService.java#L172
你觉得呢？

Reply:
在`com.ctrip.framework.apollo.biz.service.AdminService#deleteApp`确实不需要再删除branch cluster了，因为已经被删除了。
https://github.com/ctripcorp/apollo/blob/79e94e715b82f90d9ce82e14621e007253890bc4/apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/AdminService.java#L62

至于为什么`ClusterService#delete() 中的 clusterRepository.findById(id).orElse(null);这个方法能查出 isDelete=true 的数据`，看起来是因为处于同一个事物导致的，不过多删一次也没啥副作用。